csv/csv.cc

```diff
@@ -0,0 +1,100 @@
+#include <arrow/api.h>
+#include <arrow/csv/api.h>
+#include <arrow/io/api.h>
+
+#include <memory>
+
+#include "_cgo_export.h"
+#include "csv.h"
+
+// TODO: Unite with the one in carrow.cc
+struct Table {
+  std::shared_ptr<arrow::Table> table;
+};
+
+
+class GoStream: virtual public arrow::io::InputStream {
```

Comment:
Why virtual inheritance ? isn't it just for multiple inheritance cases ?

Reply:
Not sure about `virtual`, that's how I saw it in Python IIRC